### PR TITLE
Updated documentation to avoid hp::DoFHandler.

### PR DESCRIPTION
**Pull request by marcfehling**
While browsing through the documentation, I've found duplicates of functions in header files that I removed.

So although this PR effectively just changes the documentation, please run the testsuite to be sure nothing breaks.

### CODE REVIEW

**Comment by marcfehling** [comment]
Part of #10333.

File: include/deal.II/dofs/dof_tools.h [hunk 1]
```diff
@@ -2667,8 +2646,8 @@ namespace DoFTools
    */
   DeclException0(ExcGridsDontMatch);
   /**
-   * The ::DoFHandler or hp::DoFHandler was not initialized with a finite
-   * element. Please call DoFHandler::distribute_dofs() etc. first.
+   * The ::DoFHandler was not initialized with a finite element. Please call
```
Comment (drwells) [inline include/deal.II/dofs/dof_tools.h:2649]:
```suggestion
   * The DoFHandler was not initialized with a finite element. Please call
```

File: include/deal.II/distributed/solution_transfer.h [hunk 2]
```diff
@@ -243,10 +233,10 @@ namespace parallel
       /**
        * Constructor.
        *
-       * @param[in] dof The DoFHandler or hp::DoFHandler on which all
-       *   operations will happen. At the time when this constructor
-       *   is called, the DoFHandler still points to the triangulation
-       *   before the refinement in question happens.
+       * @param[in] dof The DoFHandler on which all operations will happen.
+       *   At the time when this constructor is called, the DoFHandler still
+       *   points to the triangulation before the refinement in question
```
Comment (drwells) [inline include/deal.II/distributed/solution_transfer.h:238]:
```suggestion
       *   points to the Triangulation before the refinement in question
```

File: include/deal.II/distributed/solution_transfer.h [hunk 3]
```diff
@@ -153,40 +153,31 @@ namespace parallel
...
-     * data will be packed on the parent cell with its `active_fe_index` and
-     * unpacked later with the same index on its children. If cells get
-     * coarsened into one, data will be packed on the children with the least
-     * dominant finite element of their common subspace, and unpacked on the
-     * parent with this particular finite element (consult
+     * <h3>Note on usage with DoFHandler with hp capabilities</h3>
+     *
+     * Since data on DoFHandler objects with hp capabilities is associated with
+     * many different FiniteElement objects, each cell's data has to be
+     * processed with its correpsonding `future_fe_index`. Further, if
```
Comment (drwells) [inline include/deal.II/distributed/solution_transfer.h:160]:
```suggestion
     * processed with its corresponding `future_fe_index`. Further, if
```

File: include/deal.II/grid/intergrid_map.h [hunk 4]
```diff
@@ -106,7 +106,7 @@ DEAL_II_NAMESPACE_OPEN
  *
  * Note that the template parameters to this class have to be given as
  * <tt>InterGridMap<DoFHandler<2> ></tt>, which here is DoFHandler (and could
- * equally well be Triangulation, PersistentTriangulation, or hp::DoFHandler).
+ * equally well be Triangulation, or PersistentTriangulation).
```
Comment (drwells) [inline include/deal.II/grid/intergrid_map.h:109]:
```suggestion
 * equally well be Triangulation or PersistentTriangulation).
```

File: include/deal.II/grid/grid_tools.h [hunk 5]
```diff
@@ -2180,12 +2176,12 @@ namespace GridTools
   /*@{*/
 
   /**
-   * Given two meshes (i.e. objects of type Triangulation, DoFHandler, or
-   * hp::DoFHandler) that are based on the same coarse mesh, this function
-   * figures out a set of cells that are matched between the two meshes and
-   * where at most one of the meshes is more refined on this cell. In other
-   * words, it finds the smallest cells that are common to both meshes, and
-   * that together completely cover the domain.
+   * Given two meshes (i.e. objects of type Triangulation, or DoFHandler) that
```
Comment (drwells) [inline include/deal.II/grid/grid_tools.h:2179]:
```suggestion
   * Given two meshes (i.e. objects of type Triangulation or DoFHandler) that
```

File: include/deal.II/grid/grid_tools.h [hunk 6]
```diff
@@ -1636,8 +1633,7 @@ namespace GridTools
    *
    * @tparam MeshType A type that satisfies the requirements of the
    * @ref ConceptMeshType "MeshType concept".
-   * @param mesh A mesh (i.e. objects of type Triangulation, DoFHandler,
-   * or hp::DoFHandler).
+   * @param mesh A mesh (i.e. objects of type Triangulation, or DoFHandler).
```
Comment (drwells) [inline include/deal.II/grid/grid_tools.h:1636]:
```suggestion
   * @param mesh A mesh (i.e. objects of type Triangulation or DoFHandler).
```

File: include/deal.II/grid/grid_tools.h [hunk 7]
```diff
@@ -1598,8 +1596,7 @@ namespace GridTools
...
```
Comment (drwells) [inline include/deal.II/grid/grid_tools.h:1599]:
```suggestion
   * @param mesh A mesh (i.e. objects of type Triangulation or DoFHandler).
```

File: include/deal.II/grid/grid_tools.h [hunk 8]
```diff
@@ -1558,8 +1557,7 @@ namespace GridTools
    *
    * @tparam MeshType A type that satisfies the requirements of the
    * @ref ConceptMeshType "MeshType concept".
-   * @param[in] mesh A mesh (i.e. objects of type Triangulation, DoFHandler,
-   * or hp::DoFHandler).
+   * @param[in] mesh A mesh (i.e. objects of type Triangulation, or DoFHandler).
```
Comment (drwells) [inline include/deal.II/grid/grid_tools.h:1560]:
```suggestion
   * @param[in] mesh A mesh (i.e. objects of type Triangulation or DoFHandler).
```

File: include/deal.II/grid/grid_tools.h [hunk 9]
```diff
@@ -1517,8 +1517,7 @@ namespace GridTools
...
```
Comment (drwells) [inline include/deal.II/grid/grid_tools.h:1520]:
```suggestion
   * @param[in] mesh A mesh (i.e. objects of type Triangulation or DoFHandler).
```

**Review by drwells — CHANGES_REQUESTED** [review]
Nice! I caught a few minor things (mostly with commas).

**Review by drwells — APPROVED** [review]

**Review by peterrum — APPROVED** [review]

**Pull request merged**